src/components/Box/Box.jsx

```diff
@@ -0,0 +1,598 @@
+import React, {useRef, forwardRef} from "react";
+import PropTypes from "prop-types";
+import cx from "classnames";
+import useMergeRefs from "hooks/useMergeRefs";
+import {
+    OPACITY,
+    OVERFLOW,
+    OVERFLOW_X,
+    OVERFLOW_Y,
+    BORDER,
+    BORDER_TOP,
+    BORDER_END,
+    BORDER_BOTTOM,
+    BORDER_START,
+    BORDER_COLOR,
+    ROUNDED,
+    ROUNDED_TOP,
+    ROUNDED_END,
+    ROUNDED_BOTTOM,
+    ROUNDED_START,
+    SHADOW,
+    MARGIN,
+    MARGIN_X,
+    MARGIN_Y,
+    MARGIN_TOP,
+    MARGIN_END,
+    MARGIN_BOTTOM,
+    MARGIN_START,
+    MARGIN_NEGATIVE,
+    MARGIN_NEGATIVE_X,
+    MARGIN_NEGATIVE_Y,
+    MARGIN_NEGATIVE_TOP,
+    MARGIN_NEGATIVE_END,
+    MARGIN_NEGATIVE_BOTTOM,
+    MARGIN_NEGATIVE_START,
+    PADDING,
+    PADDING_X,
+    PADDING_Y,
+    PADDING_TOP,
+    PADDING_END,
+    PADDING_BOTTOM,
+    PADDING_START,
+    TEXT_ALIGN,
+    TEXT_TRANSFORM,
+    USER_SELECT,
+    POINTER_EVENTS,
+    VISIBILITY,
+    BACKGROUND_COLORS,
+    COLORS,
+} from "./BoxConstants"
+import styles from "./Box.module.scss";
+
+const Box = forwardRef(
+    ({
+         className,
+         id,
+         children,
+         opacity,
+         overflow,
+         overflowX,
+         overflowY,
+         border,
+         borderTop,
+         borderEnd,
+         borderBottom,
+         borderStart,
+         borderColor,
+         rounded,
+         roundedTop,
+         roundedEnd,
+         roundedBottom,
+         roundedStart,
+         shadow,
+         margin,
+         marginX,
+         marginY,
+         marginTop,
+         marginEnd,
+         marginBottom,
+         marginStart,
+         marginNegative,
+         marginNegativeX,
+         marginNegativeY,
+         marginNegativeTop,
+         marginNegativeEnd,
+         marginNegativeBottom,
+         marginNegativeStart,
+         padding,
+         paddingX,
+         paddingY,
+         paddingTop,
+         paddingEnd,
+         paddingBottom,
+         paddingStart,
+         textAlign,
+         textTransform,
+         userSelect,
+         pointerEvent,
+         visibility,
```

Comment:
in my opinion is not needed. if component is not visible the can just not render it. or use class name (it's not save any time using our variables)

Reply:
+1

Reply:
+1
for this phase i will remove these, but i do think it part of a good utility interface.
visibility can be removed when we use react :)